<launch>

	<!-- Base Parameters -->
	<arg name="env_id" default="17DRP5sb8fy" />
	<arg name="map_path" default="/Rocket_ssd/dataset/data_litevloc/vnav_eval/matterport3d/s$(arg env_id)/merge_finalmap/" />
	<arg name="use_rviz" default="false" />
	<param name="use_sim_time" value="false" />

	<!-- Localization -->
	<node name="loc_pipeline" pkg="litevloc" type="loc_pipeline.py" output="screen"
		args="--map_path $(arg map_path)
					--query_data_path $(arg map_path)
					--image_size 512 288
					--device cuda --vpr_method cosplace --vpr_backbone ResNet18 --vpr_descriptors_dimension 256 
					--img_matcher master --pose_solver pnp 
					--config_pose_solver $(find litevloc)/python/config/dataset/matterport3d.yaml
					--global_pos_threshold 10.0
					--min_master_conf_thre 1.5
					--min_kpts_inliers_thre 200 
					--min_solver_inliers_thre 200 --viz" >
		<param name="frame_id_map" value="vloc_map" />
		<param name="child_frame_id" value="habitat_camera" />
	</node>

  <!-- Static TF -->
  <!-- Option 1: x y z yaw pitch roll (radians) frame_id child_frame_id period_in_ms -->
  <!-- Option 2: x y z qx qy qz qw frame_id child_frame_id period_in_ms -->
  <node pkg="tf" type="static_transform_publisher" name="vloc_trans_publisher" args="0 0 0 0 0 0 map vloc_map 10" />
	
	<!-- RVIZ -->
	<group if="$(arg use_rviz)">
		<node name="rviz" pkg="rviz" type="rviz" args="-d $(find litevloc)/rviz_cfg/topo_graph_matterport3d.rviz" required="false" />
	</group>

</launch>
